ref.go

```diff
@@ -95,13 +99,17 @@ func (e *Encoder) checkRefMap(v reflect.Value) (int, bool) {
 			v = v.Elem()
 		}
 		kind = v.Elem().Kind()
+		if kind != reflect.Invalid {
```

Comment:
when will kind be Invalid?

Reply:
If `v` is `nil`, then `v.Elem().Kind()` equsals `reflect.Invalid` and `v.Elem().Type()` will panic.